apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -534,7 +570,11 @@ export default NextAuth({
         const linkAccountNewUserData = { ...account, userId: newUser.id };
         await calcomAdapter.linkAccount(linkAccountNewUserData);
 
-        return true;
+        if (account.twoFactorEnabled) {
+          return `/auth/login?totp=${await signJwt({ email: newUser.email })}`;
```

Comment:
This is the redirection needed to let the user input a totp code in an external provider use case.